### PR TITLE
gdal/doc/source/tutorials/geotransforms_tut.rst - new tutorial with some info about geotransforms.

### PR DESCRIPTION
**Pull request by idanmiara**
## What does this PR do?
adds some documentation about geotransforms

gdal/doc/source/tutorials/geotransforms_tut.rst - new tutorial with some info about geotransforms.
gdal/doc/source/tutorials/[index.rst, raster_api_tut.rst] - reference geotransforms_tut.rst

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: gdal/doc/source/tutorials/geotransforms_tut.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+Geotransform Tutorial
+================================================================================
+
+Introduction to Geotransforms:
+-------------------------------
+
+0 = x-coordinate of the upper-left corner of the upper-left pixel
+1 = w-e pixel resolution / pixel width,
+2 = row rotation (typically zero)
+3 = y-coordinate of the of the upper-left corner of the upper-left pixel
```
Comment (rouault) [inline gdal/doc/source/tutorials/geotransforms_tut.rst:13]:
```suggestion
3 = y-coordinate of the upper-left corner of the upper-left pixel
```

File: gdal/doc/source/tutorials/geotransforms_tut.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+4 = column rotation (typically zero)
+5 = n-s pixel resolution / pixel height (typically negative value)
```
Comment (rouault) [inline gdal/doc/source/tutorials/geotransforms_tut.rst:15]:
```suggestion
5 = n-s pixel resolution / pixel height (negative value for a north-up image)
```

File: gdal/doc/source/tutorials/geotransforms_tut.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.. _geotransforms_tut:
+
+================================================================================
+Geotransform Tutorial
+================================================================================
+
+Introduction to Geotransforms:
+-------------------------------
+
```
Comment (rouault) [inline gdal/doc/source/tutorials/geotransforms_tut.rst:9]: A geotransform is an affine transformation from the image coordinate space (row, column) to the georeferenced coordinate space (projected or geographic coordinates). It consists in a set of 6 coefficients:

File: gdal/doc/source/tutorials/geotransforms_tut.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+Xgeo = gt(0) + Xpixel * gt(1) + Yline * gt(2)
+Ygeo = gt(3) + Xpixel * gt(4) + Yline * gt(5)
+
+In case of north up images:
+----------------------------
+(GT(2), GT(4)) coefficients are zero,
+(GT(1), GT(5)) is pixel size
+(GT(0), GT(3)) position is the top left corner of the top left pixel of the raster.
+
+Note that the pixel/line coordinates in the above are from (0.0,0.0) at the top left corner of the top left pixel
```
Comment (rouault) [inline gdal/doc/source/tutorials/geotransforms_tut.rst:26]: those 2 sentences apply to the general case and should be moved to the previous paragraph

File: gdal/doc/source/tutorials/geotransforms_tut.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+3 = y-coordinate of the of the upper-left corner of the upper-left pixel
+4 = column rotation (typically zero)
+5 = n-s pixel resolution / pixel height (typically negative value)
+
+Xgeo = gt(0) + Xpixel * gt(1) + Yline * gt(2)
+Ygeo = gt(3) + Xpixel * gt(4) + Yline * gt(5)
+
+In case of north up images:
+----------------------------
+(GT(2), GT(4)) coefficients are zero,
```
Comment (rouault) [inline gdal/doc/source/tutorials/geotransforms_tut.rst:22]: inconsistence case of gt in previous paragraph and GT here

Comment (idanmiara) [reply gdal/doc/source/tutorials/geotransforms_tut.rst:22]: fixed

File: gdal/doc/source/tutorials/geotransforms_tut.rst (hunk 4 above)
Comment (idanmiara) [reply gdal/doc/source/tutorials/geotransforms_tut.rst:26]: done

File: gdal/doc/source/tutorials/geotransforms_tut.rst (hunk 3 above)
Comment (idanmiara) [reply gdal/doc/source/tutorials/geotransforms_tut.rst:9]: added

**Comment by rouault** [comment]
Some RST formatting love would be needed.  https://github.com/talos-gis/gdal/blob/Branch_geotransforms_tut/gdal/doc/source/tutorials/geotransforms_tut.rst doesn't look very good

**Comment by idanmiara** [comment]
> Some RST formatting love would be needed. https://github.com/talos-gis/gdal/blob/Branch_geotransforms_tut/gdal/doc/source/tutorials/geotransforms_tut.rst doesn't look very good

Thanks for the pointer.
I think now it looks good.

File: gdal/doc/source/tutorials/geotransforms_tut.rst [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+| ``GT(1)`` w-e pixel resolution / pixel width.
+| ``GT(2)`` row rotation (typically zero).
+| ``GT(3)`` y-coordinate of the upper-left corner of the upper-left pixel.
+| ``GT(4)`` column rotation (typically zero).
+| ``GT(5)`` n-s pixel resolution / pixel height (negative value for a north-up image).
+
+Transformation from image coordinate space to georeferenced coordinate space:
+-------------------------------------------------------------------------------
+
+| ``X_geo = GT(0) + X_pixel | GT(1) + Y_line | GT(2)``
```
Comment (rouault) [inline gdal/doc/source/tutorials/geotransforms_tut.rst:26]: Haven't tested how it looks in .rst but the | character isn't appropriate for multiplication
```suggestion
| ``X_geo = GT(0) + X_pixel * GT(1) + Y_line * GT(2)``
```
Same for below formula

Comment (idanmiara) [reply gdal/doc/source/tutorials/geotransforms_tut.rst]: Thanks, that's a result of an incautious search replace...

File: gdal/doc/source/tutorials/geotransforms_tut.rst [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+================================================================================
+
+Introduction to Geotransforms:
+-------------------------------
+A geotransform is an affine transformation from the image coordinate space (row, column),
+also known as (pixel, line)
+to the georeferenced coordinate space (projected or geographic coordinates).
+
+A geotransform consists in a set of 6 coefficients:
+----------------------------------------------------
```
Comment (rouault) [inline gdal/doc/source/tutorials/geotransforms_tut.rst:14]: This line with dashes should probably be removed as it appears as a title line and just replaced by a line with a line-feed. Please check that the rendering is OK

```suggestion

```

**Pull request merged**